bindings/java/c/evmc-vm.c

```diff
@@ -95,11 +108,14 @@ JNIEXPORT void JNICALL Java_org_ethereum_evmc_EvmcVm_execute(JNIEnv* jenv,
     struct evmc_vm* evm = (struct evmc_vm*)(*jenv)->GetDirectBufferAddress(jenv, jevm);
     assert(evm != NULL);
     const struct evmc_host_interface* host = evmc_java_get_host_interface();
+    jobject jresult = allocateDirect(jenv, sizeof(struct evmc_result));
+    assert(jresult != NULL);
     struct evmc_result* result =
         (struct evmc_result*)(*jenv)->GetDirectBufferAddress(jenv, jresult);
     assert(result != NULL);
     *result = evmc_execute(evm, host, (struct evmc_host_context*)jcontext, (enum evmc_revision)jrev,
```

Comment:
This is still broken because it just copies the evmc_result struct, but the pointers to output are left loose. We need to move to a proper Java `Result` class, i.e. #555.